### PR TITLE
Forensics Console Fix #2

### PR DESCRIPTION
**Pull request by Sircrab**
Added a check to set_broken so that the Forensics console spits out whatever was inside of it before dying.

Also about the book that spawns with the console, due to the nature of computer repairs in which a New() call is made. Someone could potentially break their computer and fix it repeteadly to get free books. Setting the spawn of the book inside the initialize() method doesn't work either because the New() call from computer makes a call to initialize from there. This might be kind of a band-aid fix, but perhaps the book could be spawned inside the detective's room as a part of the map and not with the computer?

### CODE REVIEW

File: code/modules/detectivework/detective_work.dm [hunk 1]
```diff
@@ -86,6 +86,12 @@ var/const/FINGERPRINT_COMPLETE = 6	//This is the output of the stringpercent(pri
 	new /obj/item/weapon/book/manual/detective(get_turf(src))
 	return
 
+//Override of set_broken to spit out whatever item is inside once it breaks.
+/obj/machinery/computer/forensic_scanning/set_broken()
+	if(scanning)
+		scanning.loc = loc
```
Comment (9600bauds) [inline code/modules/detectivework/detective_work.dm]: scanning.forceMove(loc)

**Comment by Probe1** [comment]
> but perhaps the book could be spawned inside the detective's room as a part of the map and not with the computer?

This is the better solution.  There's no need for it to spawn with the console at all.

**Comment by PJB3005** [comment]
~~Er, don't computers have EJECTNOTDEL? If they do then they'll spit it out anyways if destroyed.~~

**Comment by Sircrab** [comment]
Made a new commit, now it correctly handles the ejection on stat == BROKEN|NOPOWER(3). I also changed the method to eject to use forceMove method (thanks bauds!). I also commented the line that spawns the  book with the console since it was agreed to put it on the map. However I don't know how to do this, if someone else could handle it or tell me how it would be much appreciated!

File: code/modules/detectivework/detective_work.dm [hunk 2]
```diff
@@ -3,6 +3,7 @@
 atom/var/list/suit_fibers
 
 atom/proc/add_fibers(mob/living/carbon/human/M)
+	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\datom/proc/add_fibers() called tick#: [world.time]")
```
Comment (clusterfack) [inline code/modules/detectivework/detective_work.dm]: Remove all these writepanics, we don't want nor need them anymore

**Comment by Sircrab** [comment]
Erased the commented lines that weren't used.

**Comment by PJB3005** [comment]
To add the book to the map:
1. Run `/maptools/1prepare_map.bat`.
2. Open the map files (`.dmm` extension) in Dream Maker (each map, please).
3. Place the book, it'll be in the object tree somewhere (use `crtl+click` to add an object on top of others, else you'll replace every other object).
4. Make sure to save.
5. Run `/maptools/2clean_map.bat`.
6. Commit the changed maps like you would commit any normal code file.

**Comment by Probe1** [comment]
Honestly the book isn't a priority.  It's blank!  

I said I'd add it after his PR was merged since I'm dicking with the maps.

**Comment by gbasood** [comment]
what even is this PR ew

**Comment by PJB3005** [comment]
Seems pretty dead. 

Comment if you want it reopened.

**Pull request closed without merging**